libraries/ms-common/src/stm32f0xx/adc.c

```diff
@@ -0,0 +1,218 @@
+#include <stdbool.h>
+
+#include "adc.h"
+#include "log.h"
+#include "stm32f0xx.h"
+
+#define TS_CAL1 0x1ffff7b8
+#define TS_CAL2 0x1ffff7c2
+#define VREFINT_CAL 0x1ffff7ba
+
+typedef struct ADCInterrupt {
+  ADCCallback callback;
+  void *context;
+  uint16_t reading;
+} ADCInterrupt;
+
+typedef struct ADCStatus {
+  ADCChannel current_channel;
+  bool continuous;
+} ADCStatus;
+
+static ADCInterrupt s_adc_interrupts[NUM_ADC_CHANNEL];
+static ADCStatus s_adc_status;
+
+static bool prv_channel_valid(ADCChannel adc_channel) {
+  if (adc_channel < 0 || adc_channel >= NUM_ADC_CHANNEL) {
```

Comment:
You only really need to check if it's greater than or equal to. We generally don't support negative numbers. Also, if you were to keep this as a function, this is a boolean expression.